component.json

```diff
@@ -13,12 +16,71 @@
       }
     },
     "jsonToXml": {
-      "title": "JSON to XML",
-      "main": "./lib/actions/jsonToXml.js",
+      "title": "JSON to XML (Deprecated)",
+      "main": "./lib/actions/jsonToXmlOld.js",
+      "deprecated": true,
       "description": "Takes the body of message passed into the component and converts to generic XML string",
       "metadata": {
         "in": {},
-        "out": "./lib/schemas/jsonToXml.out.json"
+        "out": "./lib/schemas/jsonToXmlOld.out.json"
+      }
+    },
+    "jsonToXmlV2": {
+      "title": "JSON to XML",
+      "main": "./lib/actions/jsonToXml.js",
+      "help": {
+        "link":  "#json-to-xml",
+        "description": "Takes the result of a JSONata expression and creates corresponding XML as either a string or an attachment"
+      },
+      "fields": {
+        "uploadToAttachment": {
+          "order": 3,
+          "label": "Upload XML as file to attachments",
+          "viewClass": "CheckBoxView"
+        },
+        "excludeXmlHeader": {
+          "order": 2,
+          "label": "Exclude XML Header/Description",
+          "viewClass": "CheckBoxView"
+        },
+        "headerStandalone": {
+          "order": 1,
+          "label": "Is the XML file standalone",
+          "viewClass": "CheckBoxView"
+        }
+      },
+      "metadata": {
+        "in": {
+          "type": "object",
+          "properties": {
+            "input": {
+              "title": "JSON to convert",
+              "type": "object",
+              "required": true,
+              "maxLength": 1000
```

Comment:
Is it enough 1000 symbols as max length of JSON input messages?

Reply:
This is taking advantage of some undocumented platform behavior.  It is currently being used by the email component: https://github.com/elasticio/email-component/blob/master/component.json#L52 When there is a large enough `maxLength` value, the mapper will cause the JSONata input to be rendered on more than one line.  Thoughts on using this functionality?

Reply:
I didn't know about this behavior.
This limitation doesn't allow to put text longer than 1000 symbols in the Integrator mode.
All other use cases is still working, like JSONata mode and mapping from previous steps.

Pros: Looks better, allows to use multi-line editor.
Cons: Integrator mode doesn't allow a text longer than 1000 symbols.